reporter/datadog_reporter.go

```diff
@@ -398,131 +529,11 @@ func (r *DatadogReporter) getPprofProfile() (profile *pprofile.Profile,
-
-				// To be compliant with the protocol generate a dummy mapping entry.
-				loc.Mapping = getDummyMapping(fileIDtoMapping, profile,
-					traceInfo.files[i])
-
-			case libpf.AbortFrame:
-				// Next step: Figure out how the OTLP protocol
-				// could handle artificial frames, like AbortFrame,
-				// that are not originate from a native or interpreted
-				// program.
-			default:
-				// Store interpreted frame information as Line message:
-				line := pprofile.Line{}
-
-				fileIDInfoLock, exists := r.frames.Get(traceInfo.files[i])
-				if !exists {
-					// At this point, we do not have enough information for the frame.
-					// Therefore, we report a dummy entry and use the interpreter as filename.
-					line.Function = createPprofFunctionEntry(funcMap, profile,
-						"UNREPORTED", frameKind.String())
-				} else {
-					fileIDInfo := fileIDInfoLock.RLock()
-					si, exists := (*fileIDInfo)[traceInfo.linenos[i]]
-					if !exists {
-						// At this point, we do not have enough information for the frame.
-						// Therefore, we report a dummy entry and use the interpreter as filename.
-						// To differentiate this case with the case where no information about
-						// the file ID is available at all, we use a different name for reported
-						// function.
-						line.Function = createPprofFunctionEntry(funcMap, profile,
-							"UNRESOLVED", frameKind.String())
-					} else {
-						line.Line = int64(si.lineNumber)
-
-						line.Function = createPprofFunctionEntry(funcMap, profile,
-							si.functionName, si.filePath)
-					}
-					fileIDInfoLock.RUnlock(&fileIDInfo)
-				}
-				loc.Line = append(loc.Line, line)
-
-				// To be compliant with the protocol generate a dummy mapping entry.
-				loc.Mapping = getDummyMapping(fileIDtoMapping, profile, traceInfo.files[i])
-			}
-			sample.Location = append(sample.Location, loc)
-		}
-
-		execPath, _ := r.execPathes.Get(traceKey.pid)
-
-		// Check if the last frame is a kernel frame.
-		if len(traceInfo.frameTypes) > 0 &&
-			traceInfo.frameTypes[len(traceInfo.frameTypes)-1] == libpf.KernelFrame {
-			// If the last frame is a kernel frame, we need to add a dummy
-			// location with the kernel as the function name.
-			execPath = "kernel"
-		}
-
-		if execPath != "" {
-			base := path.Base(execPath)
-			loc := createPProfLocation(profile, 0)
-			m := createPprofFunctionEntry(funcMap, profile, base, execPath)
-			loc.Line = append(loc.Line, pprofile.Line{Function: m})
-			sample.Location = append(sample.Location, loc)
-		}
-
-		sample.Label = make(map[string][]string)
-		addTraceLabels(sample.Label, traceKey)
-
-		count := int64(len(traceInfo.timestamps))
-		sample.Value = append(sample.Value, count, count*int64(r.samplingPeriod))
-		profile.Sample = append(profile.Sample, sample)
-		totalSampleCount += len(traceInfo.timestamps)
+		sample := &pprofile.Sample{}
+		count := len(traceInfo.timestamps)
+		r.processSample(sample, profile, &traceKey, traceInfo, fileIDtoMapping,
```

Comment:
I will silence this if we agree it is not an issue.

Reply:
I think this is not an issue thanks to https://go.dev/blog/loopvar-preview, and since go.mod has 1.22+ as the version https://github.com/DataDog/otel-profiling-agent/blob/main/go.mod#L3 this should be alright